### ISSUE TITLE
import great_expectations imports IPython, causes poor include performance

### ISSUE THREAD

**Issue by schrockn** [fivetran/great_expectations#573]
A no-op include of great_expectations incurs significant overhead:

```
(dagster-dev-3.7.1) [schrockn@mbp ~/code/dagster (new-ge-prototype *)]$ time python3 -c "import great_expectations"

real	0m2.839s
user	0m2.984s
sys	0m0.394s
```

A quick profile indicates that it's because IPython is included.

```
   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
   2165/1    0.012    0.000    3.330    3.330 <frozen importlib._bootstrap>:978(_find_and_load)
   2163/1    0.008    0.000    3.330    3.330 <frozen importlib._bootstrap>:948(_find_and_load_unlocked)
   1741/1    0.007    0.000    3.329    3.329 <frozen importlib._bootstrap>:663(_load_unlocked)
   1594/1    0.005    0.000    3.329    3.329 <frozen importlib._bootstrap_external>:722(exec_module)
   2579/1    0.001    0.000    3.329    3.329 <frozen importlib._bootstrap>:211(_call_with_frames_removed)
   2225/1    0.039    0.000    3.329    3.329 {built-in method builtins.exec}
        1    0.000    0.000    3.329    3.329 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/__init__.py:1(<module>)
        1    0.000    0.000    3.329    3.329 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/util.py:1(<module>)
  950/119    0.002    0.000    3.006    0.025 {built-in method builtins.__import__}
        1    0.000    0.000    1.731    1.731 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/data_context/__init__.py:3(<module>)
        1    0.000    0.000    1.730    1.730 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/data_context/data_context.py:3(<module>)
        1    0.000    0.000    1.383    1.383 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/data_context/expectation_explorer.py:1(<module>)
        1    0.000    0.000    1.286    1.286 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/IPython/__init__.py:6(<module>)
        1    0.000    0.000    1.253    1.253 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/IPython/terminal/embed.py:4(<module>)
        1    0.000    0.000    1.180    1.180 /Users/schrockn/venvs/dagster-dev-3.7.1/lib/python3.7/site-packages/great_expectations/dataset/__init__.py:1(<module>)
```

**Comment by abegong** [comment]
Yeah, that slow boot has been driving me nuts. @jcampbell was looking into it.

I believe that we need to move the `expectation_explorer.py` module out of the `data_context/` and into `jupyter_ux/` so that it'll only be imported in the context of a notebook.

@roblim Can you verify?

**Comment by eugmandel** [comment]
Thank you @schrockn ! Importing of IPython is now conditional and is not done by default - the fix will be rolled out in 0.7.7 on 2019-08-19

**Comment by schrockn** [comment]
thanks!
